nwb_conversion_tools/datainterfaces/icephys/baseicephysinterface.py

```diff
@@ -0,0 +1,145 @@
+"""Author: Luiz Tauffer."""
+from abc import ABC
+from typing import Optional
+
+from pynwb import NWBFile
+from pynwb.device import Device
+from pynwb.icephys import IntracellularElectrode
+
+from ...basedatainterface import BaseDataInterface
+from ...tools.neo import (
+    get_number_of_electrodes,
+    get_number_of_segments,
+    write_neo_to_nwb,
+)
+from ...utils import (
+    OptionalFilePathType,
+    get_schema_from_hdmf_class,
+    get_schema_from_method_signature,
+    get_base_schema,
+)
+
+
+try:
+    from ndx_dandi_icephys import DandiIcephysMetadata
+
+    HAVE_NDX_DANDI_ICEPHYS = True
+except ImportError:
+    HAVE_NDX_DANDI_ICEPHYS = False
+
+
+class BaseIcephysInterface(BaseDataInterface, ABC):
+    """Primary class for all intracellular NeoInterfaces."""
+
+    neo_class = None
+
+    @classmethod
+    def get_source_schema(cls):
+        source_schema = get_schema_from_method_signature(class_method=cls.__init__, exclude=[])
+        return source_schema
+
+    def __init__(self, file_paths: list):
+        super().__init__(file_paths=file_paths)
+
+        self.readers_list = list()
+        for f in file_paths:
+            self.readers_list.append(self.neo_class(filename=f))
+
+        self.subset_channels = None
+        self.n_segments = get_number_of_segments(neo_reader=self.readers_list[0], block=0)
+        self.n_channels = get_number_of_electrodes(neo_reader=self.readers_list[0])
+
+    def get_metadata_schema(self):
+        metadata_schema = super().get_metadata_schema()
+
+        metadata_schema["properties"]["Icephys"] = get_base_schema(tag="Icephys")
+        metadata_schema["properties"]["Icephys"]["required"] = ["Device", "IntracellularElectrode"]
```

Comment:
Is there a reason this needs to be `IntracellularElectrode` instead of just `Electrodes` if it is already in the `Icephys` key? Just trying to reduce the amount of redundant typing someone would need to do to specify this. And also to keep consistency with `Ecephys`.

Reply:
done